code/modules/urist/modules/newtrading/NPC/npc_interact.dm

```diff
@@ -7,23 +7,29 @@
 	attack_hand(usr)
 
 /mob/living/simple_animal/passive/npc/attack_hand(mob/living/user)
-	if(user && istype(user) && can_use(user))
+	if(src.stat != CONSCIOUS)
+		to_chat(user, "[src] doesn't look like he's in any shape to trade right now...")
+		interacting_mob = null
+		return
 
-		if(interacting_mob && !can_use(interacting_mob))
-			interacting_mob = null
+	if(!(user && istype(user) && can_use(user)))
```

Comment:
A lot of this is unnecessary. `can_use` already performs an `istype` check for user, which would also return null if `user` is null. If we switch the check order in `can_use` ever so slightly, it'll become null safe and allow us to skip a lot of the doubled up checks
```suggestion
	if(!can_use(user))
```